src/emulator/storage/files.ts

```diff
@@ -291,11 +295,28 @@ export class StorageLayer {
     const file = new StoredFile(finalMetadata, filePath);
     this._files.set(filePath, file);
 
+    return { upload, file };
+  }
+
+  public persistUpload(finalizedUpload: FinalizedUpload): void {
+    const { upload, file } = finalizedUpload;
+    const filePath = this.path(upload.bucketId, upload.objectId);
+
     this._persistence.deleteFile(filePath, true);
```

Comment:
optional nit: I know this is pre-existing code and its not java but consider adding an argument comment here to disambiguate the boolean arg, go/java-practices/boolean-parameters#comments

e.g. `this._persistence.deleteFile(filePath, /* failSilently = */ true)` is a lot easier to parse

Reply:
Done.